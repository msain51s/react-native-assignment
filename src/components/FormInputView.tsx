import React, { Dispatch, SetStateAction } from "react";
import { KeyboardTypeOptions, TextInput } from 'react-native'
import styles from "../utility/Styles";

type FormInputProps ={
    hint:string,
    keyboardType:KeyboardTypeOptions,
    setText:Dispatch<SetStateAction<string>>
}
export default function FormInputView(props:FormInputProps){
    return(
        <TextInput 
        style= {styles.textInputStyle}
        placeholder={props.hint}
        keyboardType={props.keyboardType}
        onChangeText={(text) => {props.setText(text)}}
        />
    )
}