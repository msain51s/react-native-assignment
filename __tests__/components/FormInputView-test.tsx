import React, { useState } from "react";
import renderer from 'react-test-renderer'
import FormInputView from "../../src/components/FormInputView";

test('FormInputView render correctly',()=>{
    let setText:any
    const tree = renderer.create(<FormInputView hint="Name" keyboardType= 'default' setText={setText}/>)
    expect(tree).toMatchSnapshot()
    expect(tree.root.props.hint).toBe('Name')
})

test('FormInputView props hint matching',()=>{
    let setText:any
    const tree = renderer.create(<FormInputView hint="Name" keyboardType= 'default' setText={setText}/>)
    expect(tree.root.props.hint).toBe('Name')
})