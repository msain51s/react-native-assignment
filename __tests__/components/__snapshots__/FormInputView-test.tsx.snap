// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputView render correctly 1`] = `
<TextInput
  keyboardType="default"
  onChangeText={[Function]}
  placeholder="Name"
  style={
    {
      "backgroundColor": "white",
      "borderColor": "grey",
      "borderRadius": 20,
      "borderWidth": 1,
      "color": "black",
      "margin": 10,
      "padding": 10,
    }
  }
/>
`;
